Assets/RGLUnityPlugin/Scripts/LidarSensor.cs

```diff
@@ -246,6 +264,44 @@ private void ApplyConfiguration(BaseLidarConfiguration newConfig)
                 rglGraphLidar.SetActive(snowNodeId, LidarSnowManager.Instance.IsSnowEnabled);
             }
 
+            // Rain model updates
+            if (rglGraphLidar.HasNode(rainNodeId))
+            {
+                // Update rain parameters only if feature is enabled (expensive operation)
+                if (LidarRainManager.Instance.IsRainEnabled)
+                {
+                    rglGraphLidar.UpdateNodePointsSimulateRain(rainNodeId,
+                        newConfig.GetRayRanges()[0].x,
+                        newConfig.GetRayRanges()[0].y,
+                        LidarRainManager.Instance.RainRate,
+                        newConfig.laserArray.GetLaserRingIds().Length,
+                        newConfig.horizontalBeamDivergence * Mathf.Deg2Rad,
+                        LidarRainManager.Instance.RainNumericalThreshold,
+                        LidarRainManager.Instance.OccupancyThreshold);
+                    rglGraphLidar.UpdateNodePointsRainDefaults(rainNodeId,
+                        LidarRainManager.Instance.DropletsId,
+                        LidarRainManager.Instance.FullBeamIntensity,
+                        0.0f); // Default, because it is not supported in AWSIM.
+                }
+
+                rglGraphLidar.SetActive(rainNodeId, LidarRainManager.Instance.IsRainEnabled);
+            }
+
+            // Fog model updates
+            if (rglGraphLidar.HasNode(fogNodeId))
+            {
+                // Update fog parameters only if feature is enabled (expensive operation)
+                if (LidarFogManager.Instance.IsFogEnabled)
+                {
+                    rglGraphLidar.UpdateNodePointsSimulateFog(fogNodeId,
+                        LidarFogManager.Instance.AttenuationCoefficient,
+                        LidarFogManager.Instance.nearCrossPoint,
+                        LidarFogManager.Instance.farCrossPoint);
```

Comment:
Whe you add a call for configuring fog simulation defaults, please also remember to add `.UpdateNodePointsRainDefaults` call below this line.